plugin/evm/vm_test.go

```diff
@@ -2076,86 +2076,96 @@ func TestBuildSubnetEVMBlock(t *testing.T) {
-	}
+			txErrors := tvm.vm.txPool.AddRemotesSync([]*types.Transaction{signedTx})
+			for i, err := range txErrors {
+				require.NoError(t, err, "Failed to add tx at index %d: %s", i, err)
+			}
 
-	txs := make([]*types.Transaction, 10)
-	for i := 0; i < 10; i++ {
-		tx := types.NewTransaction(uint64(i), testEthAddrs[0], big.NewInt(10), 21000, big.NewInt(testMinGasPrice*3), nil)
-		signedTx, err := types.SignTx(tx, types.NewEIP155Signer(tvm.vm.chainConfig.ChainID), testKeys[1].ToECDSA())
-		if err != nil {
-			t.Fatal(err)
-		}
-		txs[i] = signedTx
-	}
-	errs := tvm.vm.txPool.AddRemotesSync(txs)
-	for i, err := range errs {
-		if err != nil {
-			t.Fatalf("Failed to add tx at index %d: %s", i, err)
-		}
-	}
+			blk := issueAndAccept(t, tvm.vm)
+			newHead := <-newTxPoolHeadChan
+			require.Equal(t, newHead.Head.Hash(), common.Hash(blk.ID()), "Expected new block to match")
 
-	blk = issueAndAccept(t, tvm.vm)
-	ethBlk := blk.(*chain.BlockWrapper).Block.(*wrappedBlock).ethBlock
-	if customtypes.BlockGasCost(ethBlk) == nil || customtypes.BlockGasCost(ethBlk).Cmp(big.NewInt(100)) < 0 {
-		t.Fatalf("expected blockGasCost to be at least 100 but got %d", customtypes.BlockGasCost(ethBlk))
-	}
-	chainConfig := params.GetExtra(tvm.vm.chainConfig)
-	minRequiredTip, err := customheader.EstimateRequiredTip(chainConfig, ethBlk.Header())
-	if err != nil {
-		t.Fatal(err)
-	}
-	if minRequiredTip == nil || minRequiredTip.Cmp(big.NewInt(0.05*utils.GWei)) < 0 {
-		t.Fatalf("expected minRequiredTip to be at least 0.05 gwei but got %d", minRequiredTip)
-	}
+			txs := make([]*types.Transaction, 10)
+			for i := 0; i < 10; i++ {
+				tx := types.NewTransaction(uint64(i), testEthAddrs[0], big.NewInt(10), 21000, big.NewInt(testMinGasPrice*3), nil)
+				signedTx, err := types.SignTx(tx, types.NewEIP155Signer(tvm.vm.chainConfig.ChainID), testKeys[1].ToECDSA())
+				require.NoError(t, err)
+				txs[i] = signedTx
+			}
+			errs := tvm.vm.txPool.AddRemotesSync(txs)
+			for i, err := range errs {
+				require.NoError(t, err, "Failed to add tx at index %d: %s", i, err)
+			}
 
-	lastAcceptedID, err := tvm.vm.LastAccepted(context.Background())
-	if err != nil {
-		t.Fatal(err)
-	}
-	if lastAcceptedID != blk.ID() {
-		t.Fatalf("Expected last accepted blockID to be the accepted block: %s, but found %s", blk.ID(), lastAcceptedID)
-	}
+			blk = issueAndAccept(t, tvm.vm)
+			ethBlk := blk.(*chain.BlockWrapper).Block.(*wrappedBlock).ethBlock
 
-	// Confirm all txs are present
-	ethBlkTxs := tvm.vm.blockChain.GetBlockByNumber(2).Transactions()
-	for i, tx := range txs {
-		if len(ethBlkTxs) <= i {
-			t.Fatalf("missing transactions expected: %d but found: %d", len(txs), len(ethBlkTxs))
-		}
-		if ethBlkTxs[i].Hash() != tx.Hash() {
-			t.Fatalf("expected tx at index %d to have hash: %x but has: %x", i, txs[i].Hash(), tx.Hash())
-		}
+			// Verify BlockGasCost
+			blockGasCost := customtypes.BlockGasCost(ethBlk)
+			require.GreaterOrEqual(t, blockGasCost.Cmp(tt.expectMinBlockGasCost), 0,
```

Comment:
Thanks for expanding the test. While we're here, I think we should require the block gas cost to be zero in the case of Granite here. Otherwise it's not actually confirming the expected Granite behavior and may not catch a potential future regression.